File: backend/GameApp.Application/UseCases/SceneUseCases/SceneGetUseCase.cs
using GameApp.Domain.Entities;
using GameApp.Domain.ValueObjects.Scenes;

namespace GameApp.Application.UseCases.SceneUseCases;

public interface SceneGetUseCase
{
    public Task<Scene?> GetScene(Guid id);

    public Task<Scene?> GetSceneByName(SceneName name);
    
    public Task<IEnumerable<Scene>> GetAllScenes();
}

File: frontend/gameapp-frontend/src/application/usecases/scene-use-cases/scene-get-use-case.ts
import { Scene } from '../../../domain/entities/scenes/scene';
import type { SceneName } from '../../../domain/value-objects/scenes/scene-name';

export interface ISceneGetUseCase {
  getScene(id: string): Promise<Scene>;
  getSceneByName(name: SceneName): Promise<Scene>;
  getAllScenes(): Promise<Scene[]>;
}
